src/components/App.tsx

```diff
@@ -1,15 +1,38 @@
+import { useState } from 'react'
+import { Modal } from './modal'
+import CustomCloseIcon from './modal/CustomCloseIcon'
 
 const App = () => {
+  const [showModal, setShowModal] = useState(false)
+
+  const handleOpen = () => { setShowModal(true) }
+  const handleClose = () => { setShowModal(false) }
+
   return (
     <>
       <h1>Trabe interview</h1>
       <h2>Exercise 2: Modal</h2>
 
       <hr />
 
-    </>
+      <button onClick={handleOpen}>Show modal</button>
 
-  );
+      <Modal
+        showModal={showModal}
+        onClose={handleClose}
+        title='Hello Trabe'
+        CloseIcon={<CustomCloseIcon />}
+        footer='Developed by Jesús González Álvarez'
+        rounded
+      >
+        <p>
```

Comment:
You are still missing the API to close from the content. Another hint.

Suppose I want to do this: 

```
<Modal>
   <MyCustomContent />
</Modal>
```

How could I possibly close the modal from within MyCustomComponent. There are two options.

Reply:
I thought by doing this I was achieving to close the modal from the content. If you want me to change the modal state from a child component as you are proposing in that code, I would go for other solutions such as useContext or redux.
If that is correct, let me know and I will upload a new version tomorrow morning!
<img width="810" alt="Captura de pantalla 2022-03-22 a las 18 53 17" src="https://user-images.githubusercontent.com/51116828/159533275-8d738058-b412-4ec8-ab44-36303624073e.png">

Reply:
I've already sent you an email explaining everything. But yes, one solution would be to use useContext 👍.